src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -194,3 +195,29 @@ def create_metrics_timer(metric_values_timer: Sequence[MetricValueTimer], timest
                 "Verify the load on the Database and the available connection pool size.",
                 COLLECTION_INTERVAL_IN_SEC,
             )
+
+
+class ResourceCountMetricsCollector(MetricsCollector):
+    """
+    This Metric will track the number of resources (grouped by resources state).
+    """
+
+    def get_metric_name(self) -> str:
+        return "resource_count"
+
+    def get_metric_type(self) -> MetricType:
+        return MetricType.GAUGE
+
+    async def get_metric_value(
+        self, start_interval: datetime, end_interval: datetime, connection: Optional[asyncpg.connection.Connection]
+    ) -> Sequence[MetricValue]:
+        query: str = """
+            SELECT status,count(*)
```

Comment:
1. I believe all resource have to be collected per environment (oops, missed that before)
2. this counts all resources in the DB, which is useless. I think you want to know the resources in the current, active version.

Reply:
@Hugo-Inmanta I see you have taken a ticket about the metrics. To fix the issue about the environment I will add a column to the tables containing the environment_id. This will affect you I think.

Reply:
Isn't it possible to add to the query "WHERE environment = <active_env>" ?

Reply:
How would you do the aggregation afterwards? without knowing which data comes from which env?

Reply:
Ah, I didn't realize we would need to aggregate on different envs at the same time :+1: